DEVELOPMENT.md

```diff
@@ -0,0 +1,80 @@
+# Local Development Setup
+
+These instructions assume you have SDG cloned to a directory locally
+and are executing these commands within that directory.
+
+## Setup a Python virtual environment and install InstructLab
+
+```shell
+python3 -m venv --upgrade-deps venv
+. venv/bin/activate
+pip install git+https://github.com/instructlab/instructlab@main
+```
+
+At this point, you have InstructLab installed and a working `ilab`
+CLI. However, it is using a released version of SDG and not your
+locally cloned one.
+
```

Comment:
The previous section clearly identifies how to copy the whole project. Should this section explicitly show the need for a `git clone` of the sdg repo in the same virtual environment before executing the `pip install -e ...` command?

Reply:
For a user wanting to test custom SDG code, would it make a difference if the user uninstall the sublibrary `pip uninstall instructlab-sdg` before cloning and `pip install -e .`?

Reply:
No it wouldn't make a difference. The only thing that matters is that the packaged version of `instructlab-sdg` is uninstalled before installing the local one.

Reply:
I added a section describing how to fork and clone the repository, along with links to GitHub documentation for users that need more details.